pkg/controller/jobframework/setup.go

```diff
@@ -44,11 +49,11 @@ var (
 // this function needs to be called after the certs get ready because the controllers won't work
 // until the webhooks are operating, and the webhook won't work until the
 // certs are all in place.
-func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
-	return manager.setupControllers(mgr, log, opts...)
+func SetupControllers(ctx context.Context, mgr ctrl.Manager, log logr.Logger, retryInterval time.Duration, opts ...Option) error {
```

Comment:
How about making `retryInterval` an option, then we can setup a default in https://github.com/kubernetes-sigs/kueue/blob/86691f1265974b9b58634773c928804952787bdb/pkg/controller/jobframework/reconciler.go#L187 , and avoid doing any changes to main until necessary (we have a config key for it).

Reply:
Please set up a default value as well.

Reply:
The default value has been set [here](https://github.com/kubernetes-sigs/kueue/pull/2574/files#diff-dec60c4ac9a26c19509573fd87068ecb6bb784c0af528d38b48f5dc526eb4241R52). Let me know if this is okay.

Reply:
It may not be sufficient, if at some point we and up calling `SetupControllers` without the option set , we'll end up running loops `0s` sleep.